scipy/spatial/src/distance_impl.h

```diff
@@ -391,17 +404,19 @@ pdist_mahalanobis(const double *X, const double *covinv, double *dimbuf,
 
 static NPY_INLINE void
 pdist_cosine(const double *X, double *dm, npy_intp m, npy_intp n,
-             const double *norms)
+             double *norms_buff)
 {
     const double *u, *v;
     double cosine;
     npy_intp i, j;
+    
+    _row_norms(X, m, n, norms_buff);
 
     for (i = 0; i < m; i++) {
         for (j = i + 1; j < m; j++, dm++) {
             u = X + (n * i);
```

Comment:
I noticed and benchmarked. The code is equivalent. But I can move it out, no problem.